spec/unit/outputs/kafka_spec.rb

```diff
@@ -141,15 +144,16 @@
 
             # Return a custom Future that will raise an exception to simulate a Kafka send() problem.
             future = java.util.concurrent.FutureTask.new { raise org.apache.kafka.common.errors.TimeoutException.new("Failed") }
-            future.run
-            future
           else
-            m.call(*args)
```

Comment:
I believe this orphans `m`, which means we don't need to be wrapping original?

Reply:
right, we do not want to be calling the original or do wrapping for these.

sorry was a bit of in a hurry (failing unit specs caught me by surprise) and did not take a second look.
should have reviewed the left-overs before submitting for review, I've cleaned up all unnecessary wrapping.
thanks for pointing these out :green_heart: